pkg/services/app/add.go

```diff
@@ -109,20 +115,22 @@ func (a *App) Add(params AddParams) error {
 		return err
 	}
 
+	info := getAppResourceInfo(makeWegoApplication(params), clusterName)
+
 	gitProvider, err := a.gitProviderFactory(params.GitProviderToken)
 	if err != nil {
 		return err
 	}
 
 	var secretRef string
 	if SourceType(params.SourceType) == SourceTypeGit {
-		secretRef, err = a.createAndUploadDeployKey(params, params.Url, clusterName, gitProvider)
+		secretRef, err = a.createAndUploadDeployKey(info, params.DryRun, info.Spec.URL, gitProvider)
```

Comment:
might be a good idea to use info.Spec.URL within `createAndUploadDeployKey` as we are already passing in `info`

Reply:
Good catch, thanks

Reply:
We're using that parameter to select either the URL or the ConfigURL and need it to decide which one